src/fee_lab_to_nwb/scherrer_ophys/scherrerophysimagingextractor.py

```diff
@@ -23,29 +39,42 @@ def __init__(self, file_path: FilePathType):
         self._num_channels = 0
         self._channel_names = ["channel_0"]
 
-        with self.video_capture_context as vc:
+        with VideoCaptureContext(str(self.file_path)) as vc:
             self._num_frames = vc.get_movie_frame_count()
             self._image_size = vc.get_frame_shape()[:-1]
             self._sampling_frequency = vc.get_movie_fps()
 
     def get_frames(self, frame_idxs: ArrayType, channel: int = 0) -> NumpyArray:
         frames = []
         for frame_index in frame_idxs:
-            with self.video_capture_context as vc:
+            with VideoCaptureContext(str(self.file_path)) as vc:
                 rgb_frame = vc.get_movie_frame(frame_number=frame_index)
 
-            # Cast values from uint8 to uint16
-            green_color_data = rgb_frame[..., 1].astype(np.uint16)
-            blue_color_data = rgb_frame[..., 2].astype(np.uint16)
-            # Apply custom conversion to frames : green * 8 + blue / 8
-            gray_frame = (green_color_data * 8) + (blue_color_data / 8)
-            gray_frame = gray_frame.astype(np.uint16)
-
+            gray_frame = convert_rgb_frame_to_grayscale(rgb_frame)
             frames.append(gray_frame[np.newaxis, ...])
 
         concatenated_frames = np.concatenate(frames, axis=0).squeeze()
         return concatenated_frames
 
+    def get_video(
+        self,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+        channel: Optional[int] = 0,
+    ) -> np.ndarray:
+        start_frame = start_frame if start_frame is not None else 0
+        end_frame = end_frame if end_frame is not None else self.get_num_frames()
+
+        video_shape = (end_frame - start_frame,) + self._image_size
+
+        video = np.empty(shape=video_shape, dtype=np.uint16)
+        for frame_number in range(end_frame - start_frame):
```

Comment:
I think you might have to do something to the `self.video_capture_context` object to ensure it starts reading at the specified `start_frame`. I think the method is called `set_frame` or something? I'll check

Reply:
Ahh, not named quite as informatively as that...

But https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/behavior/movie/movie_utils.py#L87

is what you want here. `self.video_capture_context(frame_number=start_frame)` to ensure you start iterating from the correct starting point